### PR TITLE
OTC-10: Capitation Payment Report use data from tblCapitationPayment

### PR DESCRIPTION
**Pull request by dborowiecki**
Part of [OTC-10](https://openimis.atlassian.net/browse/OTC-10)

Procedure uspSSRSCapitationPayment is replaced with uspSSRSRetrieveCapitationPaymentReportData. Before report is generated, it is checked whether the batch run has been executed for given period and location.

### CODE REVIEW

File: IMIS/App_GlobalResources/Resource.resx [hunk 1]
```diff
@@ -5049,4 +5049,10 @@ adults/children)
 	<data name="L_PAYMENTTYPE1" xml:space="preserve">
 		<value>Payment Type</value>
 	</data>
+	<data name="M_CAPITATION_PAYMENT_NO_BATCH_RUN" xml:space="preserve">
```
Comment (dragos-dobre) [inline IMIS/App_GlobalResources/Resource.resx:5052]: Are these entries added to Lokalize?

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

**Comment by dragos-dobre** [comment]
I will wait for the call of the batch to modular before merging to develop.

**Comment by dborowiecki** [comment]
@dragos-dobre I've added calling modular batch process, can you review?

File: IMIS/ProcessBatches.aspx.vb [hunk 2]
```diff
@@ -627,11 +633,17 @@ Partial Public Class ProcessRelIndex
...
+            ElseIf rbProduct.Checked Then
                 If chkClaims.Checked = False Then Response.Redirect("Report.aspx?r=pbp")
                 If chkClaims.Checked = True Then Response.Redirect("Report.aspx?r=pbc&group=P")
+            ElseIf rbCapitation.Checked Then
+                dt = getCapitationPayment(Val(ddlRegionACC.SelectedValue), Val(ddlDistrictACC.SelectedValue),
+                                          Val(ddlProductAAC.SelectedValue), Val(ddlBatchAAC.SelectedValue))
+                If dt Is Nothing Then Exit Sub
+                Session("Report") = dt
+
+                Response.Redirect("Report.aspx?r=ca&tid=18")
```
Comment (dragos-dobre) [inline IMIS/ProcessBatches.aspx.vb:666]: tid is used for getting back to reports.aspx. As the capitation report is not in the report list, I don't think we need to add it here. Please remove.

File: IMIS/Reports.aspx.vb [hunk 3]
```diff
@@ -1641,26 +1570,22 @@ Partial Public Class Reports
                 Session("Report") = dt
                 url = "Report.aspx?r=rnw&tid=17"
             ElseIf SelectedValueID = 18 Then
-                If Not getCapitationPayment() Then Exit Sub
-                Session("Report") = dt
-                url = "Report.aspx?r=ca&tid=18"
-            ElseIf SelectedValueID = 19 Then
                 If Not getRejectedPhoto() Then Exit Sub
                 Session("Report") = dt
                 url = "Report.aspx?r=rp&tid=19"
```
Comment (dragos-dobre) [inline IMIS/Reports.aspx.vb:1575]: As you removed the cap rep, shouldn't you change the tid to 18 to match SelectedValueID? I don't understand why Exact didn't concat url with SelectedValueID.

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

File: IMIS/ProcessBatches.aspx.vb (hunk 2 above)
Comment (dborowiecki) [reply IMIS/ProcessBatches.aspx.vb]: tid removed from redireciton.

File: IMIS/Reports.aspx.vb (hunk 3 above)
Comment (dborowiecki) [reply IMIS/Reports.aspx.vb]: You're right, I changed tids for every report in which SelectedValueID has changed.

**Review by dragos-dobre — APPROVED** [review]

**Pull request merged**